autonav_ws/src/autonav_odometry/src/main.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+# filters.py from last year for the most part
+
+import json
+from types import SimpleNamespace
+from autonav_msgs.msg import MotorFeedback, GPSFeedback, Position
+from autonav_shared.types import DeviceState, SystemState, LogLevel # i am goig to beat pylance with a rock
+from particlefilter import ParticleFilter
+# from scr_msgs.msg import SystemState
+from autonav_shared.node import Node
+from enum import IntEnum
+import rclpy
+
+class FilterType(IntEnum):
+    PARTICLE_FILTER = 1
+    # ADD NEW FILTERS HERE
+
+class FiltersNodeConfig:
+    def __init__(self):
+        self.filter_type = FilterType.PARTICLE_FILTER
+
+class FiltersNode(Node):
+    def __init__(self):
+        super().__init__("autonav_odometry")
+        # init gps and position
+        self.first_gps = None
+        self.last_gps = None
+        self.latitude_length = self.declare_parameter("latitude_length", 111086.2).get_parameter_value().double_value
+        self.longitude_length = self.declare_parameter("longitude_length", 81978.2).get_parameter_value().double_value
+        # init filters
+        # IF WE IMPLEMENT MORE FILTERS ADD THEM HERE (AND ANYWHERE THAT COMES UP WHEN YOU CTRL-F "NEW FILTERS")!
+        self.pf = ParticleFilter(self.latitude_length, self.longitude_length)
+        self.config = FiltersNodeConfig()
+        self.onReset()
+
+    def config_updated(self, jsonObject):
+        self.config = json.loads(self.jdump(jsonObject), object_hook=lambda d: SimpleNamespace(**d))
+        # reinit filters after getting new config
+        self.onReset()
+
+    def init(self):
+        # [melee announcer voice] ready...
+        self.create_subscription(GPSFeedback, "/autonav/gps", self.onGPSReceived, 20)
+        self.create_subscription(MotorFeedback, "/autonav/MotorFeedback", self.onMotorFeedbackReceived, 20)
+        self.position_publisher = self.create_publisher(Position, "/autonav/position", 20)
+        # go!
+        self.set_device_state(DeviceState.OPERATING)
+
+    def onReset(self):
+        self.first_gps = None
+        self.last_gps = None
+        self.pf.init_particles()
+        # INIT NEW FILTERS HERE
+
+    def system_state_transition(self, old: SystemState, updated: SystemState):
+        # reinit filters when robot changes modes
+        if old.state != SystemState.AUTONOMOUS and updated.state == SystemState.AUTONOMOUS:
+            self.onReset()
+
+        if old.state != SystemState.MANUAL and updated.state == SystemState.MANUAL:
+            self.onReset()
+        # or gets going
+        if old.mobility == False and updated.mobility == True:
+            self.onReset()
+
+    def onGPSReceived(self, msg: GPSFeedback):
+        if msg.gps_fix == 0 and msg.is_locked == False:
```

Comment:
`msg.is_locked` does not exist anymore (see https://github.com/SoonerRobotics/autonav_software_2025/blob/feat/vectornav/autonav_ws/src/autonav_msgs/msg/GPSFeedback.msg)
just use msg.gps_fix. I can get you the vectornav PDF documentation on what the gps_fix number actually means (as I don't remember it at the moment)